sdk/main/src/impl/ASN1ECPrivateKey.cc

```diff
@@ -27,14 +27,25 @@ namespace Hedera::internal::asn1
 {
 ASN1ECPrivateKey::ASN1ECPrivateKey(const std::vector<std::byte>& bytes)
 {
-  decode(bytes);
+  if (bytes.size() >= MAX_ENCRYPTED_KEY_LENGHT)
+  {
+    throw BadKeyException("Over maximum possible input bytes for EC Key!");
+  }
+  else
+  {
+    decode(bytes);
+  }
 }
 
 std::vector<std::byte> ASN1ECPrivateKey::getKey() const
 {
   std::vector<std::byte> privateKey = get(OCTET_STRING);
-  if (privateKey.size() > ECDSA_KEY_LENGTH) // remove redundant padded bytes if any
-    privateKey = std::vector<std::byte>(privateKey.end() - ECDSA_KEY_LENGTH, privateKey.end());
+  if (privateKey.size() < EC_KEY_LENGTH)
+  {
+    throw BadKeyException("Data not decoded properly for input PEM/DER EC KEY bytes!");
+  }
+  if (privateKey.size() > EC_KEY_LENGTH) // remove redundant padded bytes if any
+    privateKey = std::vector<std::byte>(privateKey.end() - EC_KEY_LENGTH, privateKey.end());
```

Comment:
Wrap single-line `if` statement with braces

Reply:
Done. Don't know why Github doesn't mark it as Outdated?